core/trino-main/src/main/java/io/trino/operator/SplitOperatorInfo.java

```diff
@@ -52,4 +59,29 @@ public CatalogHandle getCatalogHandle()
     {
         return catalogHandle;
     }
+
+    @Override
+    public long getRetainedSizeInBytes()
+    {
+        return INSTANCE_SIZE + splitInfoSize(splitInfo);
+    }
+
+    // TODO drop when splitInfo is non-opaque
+    private long splitInfoSize(Object splitInfo)
+    {
+        return switch (splitInfo) {
+            case Map map -> {
+                try {
+                    yield estimatedSizeOf(map,
+                            key -> estimatedSizeOf((String) key),
+                            value -> estimatedSizeOf((String) value));
+                }
+                catch (ClassCastException e) {
+                    yield 0;
```

Comment:
having this exception-driven can burn CPU
and we still misaccount stuff.
the `splitInfo` is json serializable, so maybe serialize and check length?